### ISSUE TITLE
Changelog has gone missing

### ISSUE THREAD

**Issue by adamchainz** [pycontribs/jira#576]
Can't tell what changed in version 1.0.15 :(

**Comment by adamchainz** [comment]
No the literal changelog file for this project

**Comment by ssbarnea** [comment]
Yes you can, look at git history between tags. We will not have a in-repo changelog as it requires extra maintenance. Still, the CI testing and release process (done by travis) is supposed to produce it during build and to upload it to github release notes and on pypi. I am not sure if the github part was implemented fully or not, yet.

Lately Travis gave us lots of headaches with bugs l, for exampl the 2.0.0rc1 tag I pushed failed to publish the wheels. 

Keep in mind that this project has zero corporate support at this moment, is just the free time of few volunteers. Still, we always welcome new PRs.

**Comment by adamchainz** [comment]
Hi thanks, I understand it's by volunteers. Thanks for your effort.

Having a changelog is important for end users. If you keep it you'll have a place to communicate to users breaking changes etc., something that can't be done by reading the git logs.

P.S. imo publishing from Travis isn't worth it, controlling releases locally lets you ensure they go up the way you really want. Plus I gpg-sign my packages and my gpg key ain't leaving my control :)

**Comment by AnneTheAgile** [comment]
1.Even the git messages are not helpful due to file not being .txt and downloading, have to then change filetype, then read it. eg https://github.com/pycontribs/jira/releases/download/2.0.0/ChangeLog 
1a.I cannot see how to change it to .txt. I see that changelog is in git ignore. 
https://github.com/pycontribs/jira/search?q=changelog&unscoped_q=changelog
1b.Where / how is it created? 

2.Docs are now wrong, since they claim this file exists; 
https://github.com/pycontribs/jira/blob/4bee81141d208c2e045ed99329fcbba2801503f9/docs/index.rst#L33
says 
https://github.com/pycontribs/jira/blob/master/CHANGELOG which is generated based on git commit messages.
2a.I'm not sure PR to just delete that is good enough, [1] likely needs an answer.

3.Maybe one way would be to show a list of all PR's and issues that were closed since the prior release ? 
https://github.com/pycontribs/jira/issues?q=is%3Aissue+is%3Aclosed
https://github.com/pycontribs/jira/pulls?q=is%3Apr+is%3Aclosed
3a.The pulls have pretty good titles.

4.I totally thank you for working spare time on this. At the same time, I might stop since I have a number of issues and I am not sure how to start debugging. I am not sure how to see what changed between my version vs yours I do not understand being no reasonable view of changelog from mine. That would be sad. crossFingers.
````
$ pip show jira
Name: jira
Version: 1.0.10
Summary: Python library for interacting with JIRA via REST APIs.
Home-page: https://github.com/pycontribs/jira
Author: Sorin Sbarnea
Author-email: sorin.sbarnea@gmail.com
License: BSD
Location: ~/miniconda2/envs/py27/lib/python2.7/site-packages
Requires: requests, pbr, six, defusedxml, requests-toolbelt, argparse, ordereddict, requests-oauthlib, setuptools
Required-by: 
````

**Comment by ssbarnea** [comment]
Indeed documentation is incorrect as it points to the removed changelog. I am not sure exactly which approach we should take so any hints would be welcomed.

The idea is that maintaining a changelog become a burden and I wanted to replace it with git changelog between tags. 

I will have a look on this when I have more time so if this is really important for you feel free to find a solution, maybe even making a PR.

**Comment by AnneTheAgile** [comment]
ty for your reply @ssbarnea ! For sure I like to help give you less work, let me ponder a PR idea.

So far today one thing I realized reading today is that if we had full sentences in commit comments' first lines, then the readability of anything we generate will be better.

Also I got to hunt a bit, and there is a project that will print out the Issues and Pulls closed for a given release, eg between tag v1.0 and v1.5. Maybe it would be enough ? 
Generate changelogs between 2 latest releases ; ; https://github.com/onmyway133/github-changelogs-maker
